src/test/scala/io/iohk/ethereum/blockchain/sync/SyncSchedulerSpec.scala

```diff
@@ -0,0 +1,248 @@
+package io.iohk.ethereum.blockchain.sync
+
+import akka.util.ByteString
+import io.iohk.ethereum.Fixtures
+import io.iohk.ethereum.blockchain.sync.StateSyncUtils.{MptNodeData, TrieProvider, checkAllDataExists}
+import io.iohk.ethereum.blockchain.sync.SyncStateScheduler.{
+  AlreadyProcessedItem,
+  CannotDecodeMptNode,
+  NotRequestedItem,
+  SchedulerState,
+  SyncResponse
+}
+import io.iohk.ethereum.db.components.{EphemDataSourceComponent, Storages}
+import io.iohk.ethereum.domain.{Address, BlockchainImpl}
+import io.iohk.ethereum.vm.Generators.genMultipleNodeData
+import org.scalatest.EitherValues
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatest.matchers.must.Matchers
+import org.scalatestplus.scalacheck.ScalaCheckPropertyChecks
+
+class SyncSchedulerSpec extends AnyFlatSpec with Matchers with EitherValues with ScalaCheckPropertyChecks {
```

Comment:
Minor - names are out of sync (_SyncSchedulerSpec_, _SyncSchedulerState_, _SyncStateScheduler_)